iplass-tools-batch/src/main/java/org/iplass/mtp/tools/batch/metadata/MetaDataExport.java

```diff
@@ -0,0 +1,648 @@
+			switch (execMode) {
+			case WIZARD :
+				LogListner consoleLogListner = getConsoleLogListner();
+				addLogListner(consoleLogListner);
+
+				//環境情報出力
+				logEnvironment();
+
+				logInfo("■Start Export Wizard");
+				logInfo("");
+
+				//Wizardの実行
+				return wizard();
+			case SILENT :
+				LogListner loggingLogListner = getLoggingLogListner();
+				addLogListner(loggingLogListner);
+
+				//環境情報出力
+				logEnvironment();
+
+				logInfo("■Start Export Silent");
+				logInfo("");
+
+				//Silentの実行
+				return silent();
+			default :
+				logError("unsupport execute mode : " + execMode);
+				return false;
+			}
+		} finally {
+			logInfo("");
+			logInfo("■Execute Result :" + (isSuccess() ? "SUCCESS" : "FAILED"));
+			logInfo("");
+		}
+
+	}
+
+	public MetaDataExport execMode(ExecMode execMode) {
+		this.execMode = execMode;
+		return this;
+	}
+
+	public MetaDataExport tenantId(Integer tenantId) {
+		this.tenantId = tenantId;
+		return this;
+	}
+
+	/**
+	 * Exportします。
+	 *
+	 * @param param Export情報
+	 * @return
+	 */
+	public boolean exportMeta(final MetaDataExportParameter param) {
+
+		setSuccess(false);
+
+		try {
+			boolean isSuccess = Transaction.required(t -> {
+
+				TenantContext tc = tcs.getTenantContext(param.getTenantId());
+				return ExecuteContext.executeAs(tc, () -> {
+					ExecuteContext.getCurrentContext().setLanguage(getLanguage());
+
+					//外部から直接呼び出された場合を考慮し、Pathを取得
+					if (param.getExportMetaDataPathList() == null) {
+						param.setExportMetaDataPathList(getMetaDataPathList(param));
+					}
+
+					if (CollectionUtil.isNotEmpty(param.getExportMetaDataPathList())) {
+						logDebug("metadata path count : " + param.getExportMetaDataPathList().size());
+
+						logInfo(rs("MetaDataExport.startExportMetaData"));
+
+						//MetaDataをtempに出力
+						File metadataFile = new File(param.getExportDir(), param.getFileName() + ".xml");
+						try (PrintWriter writer = new PrintWriter(metadataFile, "UTF-8")){
+
+							mdps.write(writer, param.getExportMetaDataPathList(), new MetaDataWriteCallback() {
+
+								@Override
+								public void onWrited(String path, String version) {
+									logInfo(rs("MetaDataExport.outputMetaData", path));
+								}
+
+								@Override
+								public boolean onWarning(String path, String message, String version) {
+									logWarn(rs("MetaDataExport.warningOutputMetaData", path));
+									logWarn(message);
+									return true;
+								}
+
+								@Override
+								public void onStarted() {
+								}
+
+								@Override
+								public void onFinished() {
+								}
+
+								@Override
+								public boolean onErrored(String path, String message, String version) {
+									logError(rs("MetaDataExport.errorOutputMetaData", path));
+									logError(message);
+									return false;
+								}
+							});
+						} catch (FileNotFoundException e) {
+							throw new SystemException(e);
+						} catch (UnsupportedEncodingException e) {
+							throw new SystemException(e);

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

Exception [SystemException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [234].  Consider adding a catch clause of this specific exception in the outer try statement.